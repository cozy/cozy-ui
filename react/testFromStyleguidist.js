import React from 'react'
import Preview from 'react-styleguidist/lib/rsg-components/Preview'
import fs from 'fs'
import chunkify from 'react-styleguidist/loaders/utils/chunkify'
import pretty from 'pretty'

/*
  Styleguidist does some black magic with AST to automatically provide
  components to the code in the README ([examples-loader]).

  We manually do that below by importing each components and binding
  them to the function that executes the README code.

  examples-loader: https://github.com/styleguidist/react-styleguidist/blob/master/loaders/examples-loader.js
 */
import ActionMenu from './ActionMenu'
import Alerter from './Alerter'
import Avatar from './Avatar'
import Badge from './Badge'
import Button from './Button'
import ButtonAction from './ButtonAction'
import Checkbox from './Checkbox'
import Field from './Field'
import Hero from './Hero'
import I18n from './I18n'
import Icon from './Icon'
import Infos from './Infos'
import Input from './Input'
import IntentHeader from './IntentHeader'
import IntentIframe from './IntentIframe'
import IntentModal from './IntentModal'
import IntentOpener from './IntentOpener'
import Label from './Label'
import ListItemText from './ListItemText'
import Media from './Media'
import Menu from './Menu'
import MidEllipsis from './MidEllipsis'
import Modal from './Modal'
import Nav from './Nav'
import Overlay from './Overlay'
import Panel from './Panel'
import Radio from './Radio'
import SelectBox from './SelectBox'
import SelectionBar from './SelectionBar'
import Sidebar from './Sidebar'
import Spinner from './Spinner'
import Tabs from './Tabs'
import Text from './Text'
import Textarea from './Textarea'
import Toggle from './Toggle'
import * as content from '../docs/fixtures/content'

// Mock error otherwise there are errors with the createStylesheet function
jest.mock('react-styleguidist/lib/rsg-components/PlaygroundError', () => {
  return function(props) {
    return <div>{props.message}</div>
  }
})

const testFromStyleguidist = (name, markdown, require) => {
  const evalInContext = a =>
    // eslint-disable-next-line no-new-func
    new Function(
      'require',
      'content',
      'ActionMenu',
      'Alerter',
      'Avatar',
      'Badge',
      'Button',
      'ButtonAction',
      'Checkbox',
      'Field',
      'Hero',
      'I18n',
      'Icon',
      'Infos',
      'Input',
      'IntentHeader',
      'IntentIframe',
      'IntentModal',
      'IntentOpener',
      'Label',
      'ListItemText',
      'Media',
      'Menu',
      'MidEllipsis',
      'Modal',
      'Nav',
      'Overlay',
      'Panel',
      'Radio',
      'SelectBox',
      'SelectionBar',
      'Sidebar',
      'Spinner',
      'Tabs',
      'Text',
      'Textarea',
      'Toggle',
      'isTesting',
      'state',
      'setState',
      `const React = require("react");
      ` + a
    ).bind(
      null,
      require,
      content,
      ActionMenu,
      Alerter,
      Avatar,
      Badge,
      Button,
      ButtonAction,
      Checkbox,
      Field,
      Hero,
      I18n,
      Icon,
      Infos,
      Input,
      IntentHeader,
      IntentIframe,
      IntentModal,
      IntentOpener,
      Label,
      ListItemText,
      Media,
      Menu,
      MidEllipsis,
      Modal,
      Nav,
      Overlay,
      Panel,
      Radio,
      SelectBox,
      SelectionBar,
      Sidebar,
      Spinner,
      Tabs,
      Text,
      Textarea,
      Toggle,
      true
    )

  const options = {
    context: {
      config: {
        compilerConfig: {
          objectAssign: 'Object.assign'
        }
      },
      codeRevision: 1337
    }
  }

  describe(name, () => {
    it('should render examples', done => {
      let doneCounter = 0
      const Readme = fs.readFileSync(markdown)
      const chunks = chunkify(Readme)
      const codes = chunks.filter(x => x.type === 'code')
      const rendered = []
      const finish = () => {
        rendered.forEach(rendered => {
          expect(rendered).toMatchSnapshot(name)
        })
        done()
      }
      codes.forEach(code => {
        const root = mount(
          <Preview code={code.content} evalInContext={evalInContext} />,
          options
        )
        requestAnimationFrame(() => {
          root.update()
          rendered.push(pretty(root.html()))
          doneCounter++
          if (doneCounter === codes.length) {
            finish()
          }
        })
      })
    })
  })
}

export default testFromStyleguidist
