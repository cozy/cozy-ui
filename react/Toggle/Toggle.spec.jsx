import testFromStyleguidist from '../testFromStyleguidist'

testFromStyleguidist('Toggle', __dirname + '/Readme.md', require)
