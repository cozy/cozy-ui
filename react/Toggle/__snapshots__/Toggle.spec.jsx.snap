// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle should render examples 1`] = `
"<div>
  <div><span class=\\"toggle\\"><input type=\\"checkbox\\" id=\\"simple-toggle\\" class=\\"checkbox\\" value=\\"on\\"><label for=\\"simple-toggle\\" class=\\"label\\"></label></span></div>
</div>"
`;

exports[`Toggle should render examples 2`] = `
"<div>
  <div><span class=\\"toggle\\"><input type=\\"checkbox\\" id=\\"simple-toggle\\" class=\\"checkbox\\" value=\\"on\\"><label for=\\"simple-toggle\\" class=\\"label\\"></label></span></div>
</div>"
`;

exports[`Toggle should render examples 3`] = `
"<div>
  <div><span class=\\"toggle\\"><input type=\\"checkbox\\" id=\\"simple-toggle\\" class=\\"checkbox\\" value=\\"on\\"><label for=\\"simple-toggle\\" class=\\"label\\"></label></span></div>
</div>"
`;

exports[`Toggle should render examples 4`] = `
"<div>
  <div><span class=\\"toggle\\"><input type=\\"checkbox\\" id=\\"simple-toggle\\" class=\\"checkbox\\" value=\\"on\\"><label for=\\"simple-toggle\\" class=\\"label\\"></label></span></div>
</div>"
`;

exports[`Toggle should render examples 5`] = `
"<div>
  <div><span class=\\"toggle\\"><input type=\\"checkbox\\" id=\\"simple-toggle\\" class=\\"checkbox\\" value=\\"on\\"><label for=\\"simple-toggle\\" class=\\"label\\"></label></span></div>
</div>"
`;
